### PR TITLE
Struct/PropMap example

### PR DESCRIPTION
**Pull request by Toqozz**
From discussion in #363.

Added an example for Struct casting.  Also includes some example code/explanation regarding `PropMap`.

I'm not sure if it's in the right place or not, but it seems to fit.  Sorry for the length, but I wanted to be cover a few issues that I ran into along the way.

### CODE REVIEW

File: dbus/examples/argument_guide.md [hunk 1]
```diff
@@ -147,7 +147,69 @@ Structs
...
+    replaces_id: u32,
+    app_icon: String,
+    summary: String,
+    body: String,
+    actions: Vec<String>,
+    hints: arg::PropMap,
+    expire_timeout: i32
+) -> Result<u32, dbus::MethodErr> {
+    // PropMap is just an alias for a HashMap:
+    // pub type PropMap = HashMap<String, Variant<Box<dyn RefArg + 'static>>>;
```
Comment (diwic) [inline dbus/examples/argument_guide.md:167]: This comment can be removed, it's already stated in the previous paragraph.

File: dbus/examples/argument_guide.md [hunk 2]
```diff
@@ -147,7 +147,69 @@ Structs
...
+
+    use dbus::arg::prop_cast;
+
+    // Retrieving a property from a PropMap of type byte (`u8`).
+    let urgency: Option<&u8> = prop_cast(&hints, "urgency");
+    //let urgency: Option<u8> = prop_cast(&hints, "urgency").copied();
+    //let urgency = prop_cast::<u8>(&hints, "urgency");    // Option<&u8>
```
Comment (diwic) [inline dbus/examples/argument_guide.md:174]: These two comments could also be removed, not sure what they are for?

**Review by diwic — COMMENTED** [review]
Looks good except for some superfluous comments. Thanks for contributing!

File: dbus/examples/argument_guide.md (hunk 2 above)
Comment (Toqozz) [reply dbus/examples/argument_guide.md:174]: Just wanted to show different ways of achieving the same thing.

**Pull request merged**